pkg/networkdevice/profile/profiledefinition/metrics.go

```diff
@@ -75,34 +77,85 @@ type SymbolConfig struct {
 	MetricType ProfileMetricType `yaml:"metric_type,omitempty" json:"metric_type,omitempty"`
 }
 
+// Clone creates a duplicate of this SymbolConfig
+func (s SymbolConfig) Clone() SymbolConfig {
+	s2 := SymbolConfig{
+		OID:              s.OID,
+		Name:             s.Name,
+		MatchValue:       s.MatchValue,
+		ScaleFactor:      s.ScaleFactor,
+		Format:           s.Format,
+		ConstantValueOne: s.ConstantValueOne,
+		MetricType:       s.MetricType,
+	}
+	if s.ExtractValue != nil {
+		s2.ExtractValue = s.ExtractValue.Copy()
+	}
+	if s.MatchPattern != nil {
+		s2.MatchPattern = s.MatchPattern.Copy()
+	}
```

Comment:
`MatchPattern` is a `*regexp.Regexp`, so this is [Regexp.Copy](https://pkg.go.dev/regexp#Regexp.Copy). 

I'm wondering if I should just not copy these, though (i.e. just do `s2.MatchPattern = s.MatchPattern`), since regexes are _almost_ immutable - the only way to mutate one in place is via the little-used `.Longest()` method, so as long as we never call that (and I don't know why we ever would) it's fine if multiple objects share the same regexp object.

(My only hesitation is that I don't entirely trust the go devs not to add more mutability methods later on, because it's exactly the sort of stupid thing this language would do; golang is all about compromising good programming patterns out of a misguided belief that it's more practical.)